crates/oxc_linter/src/rules/react/jsx_curly_brace_presence.rs

```diff
@@ -572,6 +626,36 @@ fn has_adjacent_jsx_expression_containers<'a>(
         .any(JSXChild::is_expression_container)
 }
 
+/// Html strings are not respecting encoding stuff. Transforming it should keep the original context.
+/// Replaces `hello \n world` to `hello \\n world`.
+/// When the string is wrapped in a single quote. It will be replaced with a double quote.
+fn replace_html_string_to_literal(html: &str) -> String {
+    let replaced = html.cow_replace('\\', r"\\");
+    let replaced = replaced.cow_replace('"', r#"\""#);
+    format!("{{\"{replaced}\"}}")
+}
+
+fn replace_string_literal_to_html(string: &str) -> String {
+    let replaced = string.cow_replace('\\', r"\\");
+    let replaced = replaced.cow_replace('"', r#"\""#);
+    format!("\"{replaced}\"")
+}
+
+fn remove_wrapping_quotes(wrapped_str: &str) -> &str {
+    let bytes = wrapped_str.as_bytes();
+
+    if wrapped_str.len() >= 2 {
+        let first = bytes[0];
+        let last = bytes[wrapped_str.len() - 1];
+
+        if (first == b'"' && last == b'"') || (first == b'\'' && last == b'\'') {
+            return &wrapped_str[1..wrapped_str.len() - 1];
+        }
+    }
+
+    wrapped_str
```

Comment:
consider using `fixer.codegen()` in future which might allow handling some of these cases automatically